### PR TITLE
Update todo-utils version. Fix dynamic paths in tests

### PR DESCRIPTION
**Pull request by scalvert**
Updates `@ember-template-lint/todo-utils` to latest version, fixes API changes. 

Also addresses tests that had fixtures that weren't representative of true lint results.

### CODE REVIEW

File: __tests__/acceptance/eslint-with-todo-formatter.ts [hunk 1]
```diff
@@ -152,6 +152,8 @@ describe('eslint with todo formatter', function () {
   });
 
   it('should emit errors, warnings, and todos when all of these are present and INCLUDE_TODO=1 is set', async () => {
+    jest.setTimeout(5000000);
+    debugger;
```
Comment (scalvert) [inline __tests__/acceptance/eslint-with-todo-formatter.ts:156]: Will remove once I finish fixing this test.

File: __tests__/unit/mutate-errors-to-todos-test.ts [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+import { dirSync } from 'tmp';
 import { mutateTodoErrorsToTodos } from '../../src/mutate-errors-to-todos';
 import fixtures from '../__fixtures__/fixtures';
 
 describe('mutate-errors-to-todos', () => {
   it('changes only the errors that are also present in the todo map to todos', async () => {
-    const results = fixtures.eslintWithErrors();
+    const tmp = dirSync({ unsafeCleanup: true });
+    const results = fixtures.eslintWithErrors(tmp.name);
+    debugger;
```
Comment (renatoi) [inline __tests__/unit/mutate-errors-to-todos-test.ts:10]: Make sure to remove this.

Comment (scalvert) [reply __tests__/unit/mutate-errors-to-todos-test.ts:10]: Absolutely. Hence WIP :)

**Review by renatoi — APPROVED** [review]
LGTM

**Pull request merged**